src/openforms/contrib/objects_api/validators.py

```diff
@@ -78,3 +92,52 @@ def validate_document_type_references(
 
     if errors:
         raise ValidationError(errors)
+
+
+def validate_object_ownership(submission: Submission) -> None:
+    form = submission.form
+
+    try:
+        auth_info = submission.auth_info
+    except ObjectDoesNotExist:
+        raise PermissionDenied("Cannot pass data reference as anonymous user")
+
+    object = None
+    for backend in form.registration_backends.filter(backend="objects_api"):
+        if not backend.options:
+            continue
+
+        api_group = ObjectsAPIGroupConfig.objects.filter(
+            pk=backend.options.get("objects_api_group")
+        ).first()
+        if not api_group:
+            continue
+
+        try:
+            with get_objects_client(api_group) as client:
+                try:
+                    object = client.get_object(submission.initial_data_reference)
+                    break
+                except RequestException:
+                    logger.exception(
+                        "Something went wrong while trying to retrieve "
+                        "object for initial_data_reference"
+                    )
+        except NoServiceConfigured:
+            logger.exception(
+                "Something went wrong while trying to create a client "
+                "for Objects API"
+            )
+
+    if not object:
+        # If the object cannot be found, we cannot consider the ownership check failed
+        # because it is not verified that the user is not the owner
+        logger.info(
+            "Could not find object for initial_data_reference: %s",
+            submission.initial_data_reference,
+        )
+        return
+
+    # TODO should this path be configurable?
```

Comment:
yes

Reply:
What is the right place to configure this, should it be on both registration backend and prefill options? I don't know if there is a way around that, since I think it could be possible to use one without the other

Reply:
Yeah indeed, and the UI copy button is then the convenience so that users can setup prefill from their registration configuration

Reply:
I've added a TODO in the PR to add the field to the prefill config form as well (since this is probably reliant on #4799)

Reply:
has been added